core/src/main/scala/org/apache/spark/scheduler/Stage.scala

```diff
@@ -70,6 +70,26 @@ private[spark] abstract class Stage(
   /** Pointer to the latest [StageInfo] object, set by DAGScheduler. */
   var latestInfo: StageInfo = StageInfo.fromStage(this)
 
+  // The maximum number of times to retry a stage before aborting 
+  private[scheduler] val maxStageFailures = 4
+  
+  // Spark is resilient to executors dying by retrying stages on FetchFailures. This counter ensures
+  // that we don't retry stages indefinitely by aborting the stage if it fails too many times. 
+  private var failCount = 0
+  private[scheduler] def fail() : Unit = { failCount += 1 }
+  private[scheduler] def shouldAbort(): Boolean = { failCount >= maxStageFailures }
+  private[scheduler] def clearFailures() : Unit = { failCount = 0 }
+
+  /**
+   * Check whether we should abort the failedStage due to multiple failures.
+   * This method updates the running count of failures for a particular stage and returns 
+   * true if the number of failures exceeds the allowable number of failures.
+   */
+  private[scheduler] def failAndShouldAbort(): Boolean = {
+    fail()
+    shouldAbort()
```

Comment:
can you inline those two functions here?  I don't see much value in separating them out